src/charm.py

```diff
@@ -224,37 +218,64 @@ def _on_database_storage_detaching(self, _) -> None:
     # =======================
     #  Custom Action Handlers
     # =======================
+    def _get_cluster_status(self, event: ActionEvent) -> None:
+        """Action used to retrieve the cluster status."""
+        event.set_results(self._mysql.get_cluster_status())
 
-    def _on_get_cluster_admin_credentials(self, event: ActionEvent) -> None:
-        """Action used to retrieve the cluster admin credentials."""
-        event.set_results(
-            {
-                "cluster-admin-username": CLUSTER_ADMIN_USERNAME,
-                "cluster-admin-password": self._get_secret("app", CLUSTER_ADMIN_PASSWORD_KEY),
-            }
-        )
+    def _on_get_password(self, event: ActionEvent) -> None:
+        """Action used to retrieve the system user's password."""
+        if "username" not in event.params:
+            raise RuntimeError("Undefined parameter username.")
+
+        if event.params["username"] not in REQUIRED_USERNAMES:
+            raise RuntimeError("Invalid username.")
+
+        username = event.params["username"]
```

Comment:
I would set `username=root` as the default username to make UX simpler. and the same for setting password
```suggestion
        username = ROOT_USERNAME
        if "username" in event.params:
            username = event.params["username"]

        if username not in REQUIRED_USERNAMES:
            raise RuntimeError("Invalid username.")

```